go/teams/service_helper.go

```diff
@@ -866,39 +863,35 @@ func loadUserVersionAndPUKedByUsername(ctx context.Context, g *libkb.GlobalConte
 }
 
 func loadUserVersionByUsername(ctx context.Context, g *libkb.GlobalContext, username string) (keybase1.UserVersion, error) {
-	res := g.Resolver.ResolveWithBody(username)
-	if res.GetError() != nil {
-		if e, ok := res.GetError().(libkb.ResolutionError); ok && e.Kind == libkb.ResolutionErrorNotFound {
+	m := libkb.NewMetaContext(ctx, g)
+	upk, err := engine.ResolveAndCheck(m, username)
+	if err != nil {
+		if e, ok := err.(libkb.ResolutionError); ok && e.Kind == libkb.ResolutionErrorNotFound {
 			// couldn't find a keybase user for username assertion
 			return keybase1.UserVersion{}, errInviteRequired
 		}
-		return keybase1.UserVersion{}, res.GetError()
+		return keybase1.UserVersion{}, err
 	}
 
-	return loadUserVersionByUIDCheckUsername(ctx, g, res.GetUID(), res.GetUsername())
+	return loadUserVersionByUPK2(ctx, g, upk)
 }
 
 func loadUserVersionByUID(ctx context.Context, g *libkb.GlobalContext, uid keybase1.UID) (keybase1.UserVersion, error) {
-	return loadUserVersionByUIDCheckUsername(ctx, g, uid, "")
-}
-
-func loadUserVersionByUIDCheckUsername(ctx context.Context, g *libkb.GlobalContext, uid keybase1.UID, un string) (keybase1.UserVersion, error) {
 	upak, err := loadUPAK2(ctx, g, uid, true /*forcePoll */)
 	if err != nil {
 		return keybase1.UserVersion{}, err
 	}
-	if un != "" && !libkb.NormalizedUsername(un).Eq(libkb.NormalizedUsername(upak.Current.Username)) {
-		return keybase1.UserVersion{}, libkb.BadUsernameError{N: un}
-	}
+	return loadUserVersionByUPK2(ctx, g, upak.Current)
+}
 
-	uv := NewUserVersion(upak.Current.Uid, upak.Current.EldestSeqno)
-	if upak.Current.Status == keybase1.StatusCode_SCDeleted {
+func loadUserVersionByUPK2(ctx context.Context, g *libkb.GlobalContext, upak keybase1.UserPlusKeysV2) (keybase1.UserVersion, error) {
```

Comment:
How about naming this `filterForMemberAddition`? And it doesn't need `g` as a parameter.